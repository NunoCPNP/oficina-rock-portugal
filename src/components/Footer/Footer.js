import { AiFillFacebook, AiFillInstagram, AiFillPhone, AiOutlineWhatsApp, AiTwotoneMail } from 'react-icons/ai'
import Image from 'next/image'
import { SiTiktok } from 'react-icons/si'
import useTranslation from 'next-translate/useTranslation'

import Copyright from '../Copyright'

import { ContactsContainer, GridContainer, GridItem, MethodsContainer, Spacer, StyledFooter } from './Footer.styles'

const Footer = () => {
  const { t } = useTranslation('common')

  return (
    <StyledFooter data-testid="footer">
      <GridContainer>
        <GridItem>
          <div>
            <h4>{t(`who-we-are`)}</h4>
          </div>
          <div>
            <h4>{t(`privacy-policies`)}</h4>
          </div>
          <div>
            <h4>{t(`return-and-refund`)}</h4>
          </div>
          <div>
            <h4>{t(`faq`)}</h4>
          </div>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`contacts`)}</h3>
          <ContactsContainer>
            <div>
              <AiFillPhone data-testid="react-icon" />
              <AiOutlineWhatsApp data-testid="react-icon" />
              <span>+351 964 631 183</span>
            </div>
            <div>
              <AiFillFacebook data-testid="react-icon" />
              <AiFillInstagram data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
            <div>
              <AiTwotoneMail data-testid="react-icon" />
              <span>oficinarockpt@gmail.com</span>
            </div>
            <div>
              <SiTiktok data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
          </ContactsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`payment-methods`)}</h3>
          <MethodsContainer>
            <div>
              <Image src="/payment_methods.png" alt={t(`payment-methods`)} width={160} height={35} />
            </div>
          </MethodsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`newsletter`)}</h3>
          <Spacer />
        </GridItem>
      </GridContainer>
      <Copyright />
    </StyledFooter>
  )
}

export default Footer
