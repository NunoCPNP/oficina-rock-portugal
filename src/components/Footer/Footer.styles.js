import styled from '@emotion/styled'
import mq from '@/styles/mediaQueries'

export const StyledFooter = styled.footer`
  background-color: var(--ghostWhite);
`

export const GridContainer = styled.div`
  max-width: 140rem;
  margin: 0 auto;
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;

  ${mq('m')} {
    grid-template-columns: 1fr 1fr;
  }

  ${mq('s')} {
    grid-template-columns: 1fr;
  }
`

export const GridItem = styled.div`
  line-height: 3rem;
  h3 {
    text-transform: uppercase;
    padding-bottom: 1rem;
  }

  h4 {
    font-weight: 400;
  }
`

export const ContactsContainer = styled.div`
  padding-top: 1rem;

  div {
    display: flex;
    align-items: center;
    margin-bottom: 0.4rem;

    svg {
      width: 2.2rem;
      height: 2.2rem;
      margin-right: 0.6rem;
    }
  }
`

export const MethodsContainer = styled.div`
  padding-top: 1rem;

  div {
    display: flex;
    align-items: center;
  }
`

export const Spacer = styled.div`
  display: none;
  border-bottom: 1px solid var(--darkLiver);
  margin-top: 2rem;
  opacity: 0.4;

  ${mq('s')} {
    display: block;
  }
`
