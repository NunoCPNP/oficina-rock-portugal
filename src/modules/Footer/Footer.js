import { AiFillFacebook, AiFillInstagram, AiFillPhone, AiOutlineWhatsApp, AiTwotoneMail } from 'react-icons/ai'
import Image from 'next/image'
import Link from 'next/link'
import { SiTiktok } from 'react-icons/si'
import useTranslation from 'next-translate/useTranslation'

import Copyright from '@/components/Copyright'

import {
  ContactsContainer,
  GridContainer,
  GridItem,
  Item,
  MethodsContainer,
  Spacer,
  StyledFooter,
} from './Footer.styles'

const Footer = () => {
  const { t } = useTranslation('common')

  return (
    <StyledFooter data-testid="footer">
      <GridContainer>
        <GridItem>
          <Item>
            <Link href="/whoweare" passHref>
              <h4>{t(`who-we-are`)}</h4>
            </Link>
          </Item>
          <Item>
            <Link href="/privacypolicy" passHref>
              <h4>{t(`privacy-policies`)}</h4>
            </Link>
          </Item>
          <Item>
            <Link href="/returnpolicy" passHref>
              <h4>{t(`shipping-and-exchanges`)}</h4>
            </Link>
          </Item>
          {/* <Item>
            <Link href="/faq" passHref>
              <h4>{t(`faq`)}</h4>
            </Link>
          </Item> */}
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`contacts`)}</h3>
          <ContactsContainer>
            <div>
              <AiFillPhone data-testid="react-icon" />
              <AiOutlineWhatsApp data-testid="react-icon" />
              <span>+351 964 631 183</span>
            </div>
            <div>
              <AiFillFacebook data-testid="react-icon" />
              <AiFillInstagram data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
            <div>
              <AiTwotoneMail data-testid="react-icon" />
              <span>oficinarockpt@gmail.com</span>
            </div>
            <div>
              <SiTiktok data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
          </ContactsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`payment-methods`)}</h3>
          <MethodsContainer>
            <div>
              <Image src="/payment_methods.png" alt={t(`payment-methods`)} width={160} height={35} />
            </div>
          </MethodsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`newsletter`)}</h3>
          <Spacer />
        </GridItem>
      </GridContainer>
      <Copyright />
    </StyledFooter>
  )
}

export default Footer
