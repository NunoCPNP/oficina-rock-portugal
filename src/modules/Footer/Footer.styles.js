import styled from '@emotion/styled'

export const StyledFooter = styled.footer`
  background-color: var(--ghostWhite);
`

export const GridContainer = styled.div`
  max-width: 140rem;
  margin: 0 auto;
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;
`

export const GridItem = styled.div`
  h4 {
    font-weight: 400;
    line-height: 3rem;
  }
`
