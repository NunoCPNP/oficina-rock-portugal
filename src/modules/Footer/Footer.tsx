type Props = {}

const Footer: React.FC<Props> = () => {
  return <footer></footer>
}

export default Footer