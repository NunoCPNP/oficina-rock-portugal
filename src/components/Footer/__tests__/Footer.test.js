import { render } from '@testing-library/react'

import Footer from '../Footer'

describe('<Footer /> spec', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should render with correct styling', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toHaveStyle(`
      background-color: var(--ghostWhite);
    `)
  })

  it('Should render translations keys', () => {
    const { getByText } = render(<Footer />)

    expect(getByText('common:who-we-are')).toBeInTheDocument()
    expect(getByText('common:privacy-policies')).toBeInTheDocument()
    expect(getByText('common:shipping-and-exchanges')).toBeInTheDocument()
    expect(getByText('common:faq')).toBeInTheDocument()
    expect(getByText('common:contacts')).toBeInTheDocument()
    expect(getByText('common:payment-methods')).toBeInTheDocument()
    expect(getByText('common:newsletter')).toBeInTheDocument()
  })

  it('Should render all React-Icons svg', () => {
    const { queryAllByTestId } = render(<Footer />)

    expect(queryAllByTestId(/react-icon/i)).toHaveLength(6)
  })

  it('Should render Payment Options image', () => {
    const { getAllByRole } = render(<Footer />)

    expect(getAllByRole('img')).toHaveLength(2)
  })

  it('Should render Copyright Component', () => {
    const { queryAllByTestId } = render(<Footer />)

    expect(queryAllByTestId(/copyright/i)).toHaveLength(1)
  })
})
