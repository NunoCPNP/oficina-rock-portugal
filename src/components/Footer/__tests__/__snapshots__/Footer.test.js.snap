// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> spec Should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: var(--ghostWhite);
}

.emotion-1 {
  max-width: 140rem;
  margin: 0 auto;
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;
}

.emotion-2 h4 {
  font-weight: 400;
  line-height: 3rem;
}

<footer
    class="emotion-0"
    data-testid="footer"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div>
          <h4>
            Quem Somos
          </h4>
        </div>
        <div>
          <h4>
            Políticas de Privacidade
          </h4>
        </div>
        <div>
          <h4>
            Políticas de Devolução e Reembolso
          </h4>
        </div>
        <div>
          <h4>
            Trocas e Devoluções
          </h4>
        </div>
        <div>
          <h4>
            Dúvidas Frequentes
          </h4>
        </div>
      </div>
      <div
        class="emotion-2"
      >
        <h3>
          Contactos
        </h3>
      </div>
      <div
        class="emotion-2"
      >
        <h3>
          Métodos de Pagamento
        </h3>
      </div>
      <div
        class="emotion-2"
      >
        <h3>
          Newsletter
        </h3>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
